doc/SLSKPROTOCOL.md

```diff
@@ -2777,58 +2777,62 @@ A peer sends this message when it has a file search match. The token is taken fr
         1.  **uchar** 1
         2.  **string** <ins>filename</ins>
         3.  **uint64** <ins>size</ins>
-        4.  **string** <ins>ext</ins>
-        5.  **uint** <ins>attribute size</ins>
-        6.  Iterate <ins>number of attributes</ins>
-            1.  **uint** <ins>place in
-                attributes</ins>
+        4.  **string** <ins>ext</ins> *(SoulseekNS requires "mp3" to show the attributes)*
+        5.  **uint** <ins>number of attributes</ins> 0 *or* 3
+        6.  Iterate for <ins>number of attributes</ins>
+            1.  **uint** <ins>place in attributes</ins> 0 *(bitrate)*;  1 *(duration)*;  2 *(vbr)*
             2.  **uint** <ins>attribute</ins>
+                - <ins>bitrate</ins> *Kbps*
+                - <ins>duration</ins> *Seconds*
+                - <ins>vbr</ins> 0 *or* 1 *(is Variable Bit Rate)*
   - Receive
     1.  decompress
     2.  **string** <ins>username</ins>
     3.  **uint** <ins>token</ins>
-    4.  **uint** <ins>results size</ins> *number of results*
+    4.  **uint** <ins>number of results</ins>
     5.  Iterate for <ins>number of results</ins>
         1.  **string** <ins>filename</ins>
         2.  **uint64** <ins>size</ins>
-        3.  **string** <ins>ext</ins>
-        4.  **uint** <ins>number of attributes</ins>
-        5.  Iterate <ins>number of attributes</ins>
-            1.  **uint** <ins>place in
-                attributes</ins>
+        3.  **string** <ins>ext</ins> *(Always blank from SoulseekQt clients)*
+        4.  **uint** <ins>number of attributes</ins> *(Usually 2 or 3; can be 1 in rare cases; 0 is no metadata)*
+        5.  Iterate for <ins>number of attributes</ins>
+            1.  **uint** <ins>place in attributes</ins> *(Order varies between clients and file type)*
             2.  **uint** <ins>attribute</ins>
+                - *Either bitrate or sample rate*
     6.  **bool** <ins>slotfree</ins>
     7.  **uint** <ins>avgspeed</ins>
     8.  **uint64** <ins>queue length</ins>
-    9.  **uint** <ins>private results size</ins> *number of privately shared results*
+    9.  **uint** <ins>number of privately shared results</ins>
     10.  Iterate for <ins>number of privately shared results</ins>
          1.  **string** <ins>filename</ins>
          2.  **uint64** <ins>size</ins>
-         3.  **string** <ins>ext</ins>
-         4.  **uint** <ins>number of attributes</ins>
-         5.  Iterate <ins>number of attributes</ins>
-             1.  **uint** <ins>place in
-                 attributes</ins>
+         3.  **string** <ins>ext</ins> *(Always blank from SoulseekQt clients)*
+         4.  **uint** <ins>number of attributes</ins> *(Usually 2 or 3; can be 1 in rare cases; 0 is no metadata)*
+         5.  Iterate for <ins>number of attributes</ins>
+             1.  **uint** <ins>place in attributes</ins> *(Order varies between clients and file type)*
              2.  **uint** <ins>attribute</ins>
+                 - *Either bitrate or sample rate*
```

Comment:
This doesn't belong here. We should document these attributes at a later point, preferably in a separate section.

Reply:
Yes I see several sections deal with attributes, and outlining them isn't straight-forward.

My main objective is to indicate what N+ sends, and hope that others eventually feel inspired to follow suit.